tests/Unit/CheckJwtMiddlewareTest.php

```diff
@@ -0,0 +1,260 @@
+            $this->fail();
+        } catch (HttpException $e) {
+            self::assertEquals('Insufficient scope', $e->getMessage());
+            self::assertEquals(403, $e->getStatusCode());
+        }
+    }
+
+    /**
+     * Verify the request is forbidden when the bearer token contains an empty scope, but a scope is required.
+     */
+    public function testScopeRequiredEmptyScopesProvided()
+    {
+        $this->mockAuth0Service(['scope' => '']);
+
+        $request = new Request();
+        $middleware = app()->make(CheckJWT::class);
+
+        // Make sure a bearer token is set.
+        $request->headers->set('Authorization', 'Bearer token');
+
+        try {
+            // Require 'test' scope to be present, however empty scope present in the JWT.
+            $middleware->handle($request, function () {
+                // Next middleware should not be called since there is an error with the request.
+                $this->fail();
+            }, 'test');
+
+            // Fail if no exception was thrown.
+            $this->fail();
+        } catch (HttpException $e) {
+            self::assertEquals('Insufficient scope', $e->getMessage());
+            self::assertEquals(403, $e->getStatusCode());
+        }
+    }
+
+    /**
+     * Verify the request is forbidden when the bearer token only has a different scope than required.
+     */
+    public function testScopeRequiredIncorrectScopeProvided()
+    {
+        $this->mockAuth0Service(['scope' => 'nottest']);
+
+        $request = new Request();
+        $middleware = app()->make(CheckJWT::class);
+
+        // Make sure a bearer token is set.
+        $request->headers->set('Authorization', 'Bearer token');
+
+        try {
+            // Require 'test' scope to be present, however different scope present in the JWT.
+            $middleware->handle($request, function () {
+                // Next middleware should not be called since there is an error with the request.
+                $this->fail();
+            }, 'test');
+
+            // Fail if no exception was thrown.
+            $this->fail();
+        } catch (HttpException $e) {
+            self::assertEquals('Insufficient scope', $e->getMessage());
+            self::assertEquals(403, $e->getStatusCode());
+        }
+    }
+
+    /**
+     * Verify the request passes when it contains one scope only, which is required.
+     */
+    public function testScopeRequiredOneScopeProvided()
+    {
+        $this->mockAuth0Service(['scope' => 'test']);
+
+        $request = new Request();
+        $middleware = app()->make(CheckJWT::class);
+
+        // Make sure a bearer token is set.
+        $request->headers->set('Authorization', 'Bearer token');
+
+        $nextCalled = false;
+
+        // Require 'test' scope to be present, which is present in the JWT.
+        $middleware->handle($request, function () use (&$nextCalled) {
+            $nextCalled = true;
+        }, 'test');
```

Comment:
This test file also contains quite some 'test'. I think they are all scopes, but it'd be great if you can change them all to contain some non-'test' value.